config/initializers/acts_as_favorable.rb

```diff
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
+
+# For development and non-eager load mode, we need to register the acts_as_favorable models manually
+# as no eager loading takes place
+Rails.application.config.after_initialize do
+  OpenProject::Acts::Favorable::Registry
+    .add(Project)
```

Comment:
Personally not the hugest fan of needing an initializer for this, but I can completely understand why this is needed. I also have no better ideas at the top of my head...

Maybe we can do something that when we call the `acts_as_favorable` method in a class we check that the class is included in the Registry and if not, we raise an exception that it needs to be added here in the initializer? I bet in a few weeks we all forgot that adding the class name here is required for everything to work.

Reply:
Yes that's a good point, I was also annoyed that we need something like this - but except from eager loading in dev mode, I'm not aware of how to prevent this. In a way it's weird this problem is so common and different in dev/prod mode.

Reply:
Raising an error fails because an initializer in a plugin tries to access `Project` before the acts_as_favorable initializer. I've added a commit that does the same that watchable does: Auto-registering. That doesn't solve the problem of forgetting about it in dev mode, but at least ensures it's there once autoloaded?